### PR TITLE
Add to start with matcher

### PR DESCRIPTION
**Pull request by GaryLeutheuser**
### What
Implementation of the `toStartWith` matcher.

### Why
This implementation addresses #39.

### Housekeeping

- [x] Unit tests
- [x] Documentation is up to date
- [x] No additional lint warnings
- [x] Add yourself to contributors list (`yarn contributor`)

### CODE REVIEW

**Comment by GaryLeutheuser** [comment]
I've just noticed that I haven't added an `export` to `src/matchers/index.js`, I'll do that now.

File: src/matchers/toStartWith/predicate.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export default (prefix, string) => prefix === string.substring(0, prefix.length);
```
Comment (mattphillips) [inline src/matchers/toStartWith/predicate.js]: You can also use `string.startsWith(prefix)` here. Your solution is fine too :smile:

File: src/matchers/toStartWith/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { matcherHint, printExpected, printReceived } from 'jest-matcher-utils';
+
+import predicate from './predicate';
+
+const passMessage = (prefix, string) => () =>
```
Comment (mattphillips) [inline src/matchers/toStartWith/index.js]: I think the error message would be better to show the expected too.

```js
const passMessage = (prefix, string) => () =>
  matcherHint('.not.toStartWith') +
  '\n\n' +
  'Expected string to not start with:\n' +
  `  ${printExpected(string)}`;
  'Received:\n' +
  `  ${printReceived(prefix)}`;
```

**Comment by mattphillips** [comment]
Couple of minor points. Great job! Don't forget to star and share the project :smile:

File: src/matchers/toStartWith/index.js (hunk 2 above)
Comment (GaryLeutheuser) [reply src/matchers/toStartWith/index.js]: Definitely - it's been added.

File: src/matchers/toStartWith/predicate.js (hunk 1 above)
Comment (GaryLeutheuser) [reply src/matchers/toStartWith/predicate.js]: Wow, that's really nice. I switched it over.

**Comment by mattphillips** [comment]
Nice one @GaryLeutheuser 🎉

**Pull request merged**